src/test/java/software/amazon/awssdk/crt/test/MqttClientConnectionFixture.java

```diff
@@ -65,33 +67,63 @@
     Path pathToKey = null;
     Path pathToCa = null;
 
+    String certificatePem = null;
+    String privateKeyPem = null;
+    String caRoot = null;
+    String iotEndpoint = null;
+
     protected void modifyConnectionConfiguration(MqttConnectionConfig config) {}
 
     private boolean findCredentials() {
+        CrtTestContext ctx = getContext();
+
+        // For each parameter, check the context first, then check the file system/system properties
         try {
-            pathToCert = Paths.get(TEST_CERTIFICATE);
-            pathToKey = Paths.get(TEST_PRIVATEKEY);
-            pathToCa = Paths.get(TEST_ROOTCA);
-            if (pathToCert == null || pathToCert.toString().equals("")) {
-                throw new MissingCredentialsException("Certificate not provided");
-            }
-            if (!pathToCert.toFile().exists()) {
-                throw new MissingCredentialsException("Certificate could not be found at " + pathToCert);
+            if (ctx.iotCARoot == null) {
```

Comment:
Feels like these could be folded into CrtTestContext

Reply:
They're in the MQTT fixture because they're specific to MQTT